src/main/java/io/github/externschool/planner/entity/schedule/ScheduleEvent.java

```diff
@@ -0,0 +1,319 @@
+package io.github.externschool.planner.entity.schedule;
+
+import io.github.externschool.planner.entity.User;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Danil Kuznetsov (kuznetsov.danil.v@gmail.com)
+ */
+@Entity
+@Table(name = "schedule_event")
+public class ScheduleEvent {
+
+    //TODO need to extract to base entity class and change strategy to sequence
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @Column(nullable = false)
+    private String title;
+
+    @Column(length = 5000)
+    private String description;
+
+    @Column
+    private String location;
+
+    @Column(name = "start_event", nullable = false)
+    private LocalDateTime startOfEvent;
+
+    @Column(name = "end_event", nullable = false)
+    private LocalDateTime endOfEvent;
+
+    @Column(name = "created_at", nullable = false)
+    private LocalDateTime createdAt;
+
+    @Column(name = "modified_at")
+    private LocalDateTime modifiedAt;
+
+    @Column(name = "is_open")
+    private boolean isOpen;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "owner_id")
+    private User owner;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "event_type_id")
+    private ScheduleEventType type;
+
```

Comment:
Don't you think we should add a Boolean 'occurred' field meaning the planned event has occurred, to get some statistics about students who miss their appointments? Very rare opposite cases when teachers cancel their scheduled meetings could be ignored, i guess.
or happened, accomplished (isAccomplished sounds great)

Reply:
Yes, you're right, I will fix it